snapcraft/commands/remote.py

```diff
@@ -193,7 +198,27 @@ def _run_new_or_fallback_remote_build(self, base: str) -> None:
             run_legacy()
             return
 
-        if is_repo(Path().absolute()):
+        current_path = Path().absolute()
+
+        if is_shallow_repo(current_path):
+            emit.debug("Current git repository is shallow cloned.")
+            base = self._get_effective_base()
+            if base in ["core20", "core22"]:
```

Comment:
there is already a check on line 178 for the base, so we shouldn't be checking it again with a nested if-statement